scipy/optimize/tests/test__dual_annealing.py

```diff
@@ -258,3 +259,46 @@ def test_from_docstring(self):
                          -3.93616815e-09, -6.55623025e-09, -6.05775280e-09,
                          -5.00668935e-09], atol=4e-8)
         assert_allclose(ret.fun, 0.000000, atol=5e-13)
+
+    @pytest.mark.xfail(reason='gh-10892')
+    @pytest.mark.parametrize('e', [0, 2])
+    def test_accept_reject_probabilistic(self, e):
+        # Test accepts unconditionally with e < current_energy and
+        # probabilistically with e > current_energy
+
+        rs = check_random_state(123)
+
+        count_accepted = 0
+        iterations = 1000
+
+        accept_param = -5
+        temp_step = 15
+        current_energy = 1
+        for _ in range(iterations):
+            energy_state = EnergyState(lower=None, upper=None)
+            # Set energy state with current_energy, any location.
+            energy_state.update_current(current_energy, [0])
+
+            chain = StrategyChain(
+                accept_param, None, None, None, rs, energy_state)
+            # Normally this is set in run()
+            chain.temperature_step = temp_step
+
+            # Check if new e is accepted.
+            chain.accept_reject(j=1, e=e, x_visit=[2])
+            if energy_state.current_energy == e:
+                count_accepted += 1
+
+        # Check accept rate.
+        pqv = 1 - (1 - accept_param) * (e - current_energy) / temp_step
+        rate = np.exp(np.log(pqv) / (1 - accept_param))
+
+        print ("\te:", e, "rate:", rate, "accepted:", count_accepted)
+
+        if e < current_energy:
+            assert rate >= 1
+            assert count_accepted == iterations
+        else:
+            assert_allclose(rate, 0.9183859021684453)
+            assert count_accepted == 911
```

Comment:
Currently
```
    e: 0 rate: 1.0576809264052165 accepted: 848
    e: 2 rate: 0.9183859021684453 accepted: 0
```

After we move the +1 outside of a paren in #10892 this test passes